plugin/trino-hive/src/test/java/io/trino/plugin/hive/s3/S3HiveQueryRunner.java

```diff
@@ -44,46 +44,115 @@ public static DistributedQueryRunner create(
-                        .putAll(additionalHiveProperties)
-                        .build())
-                .setInitialTables(ImmutableList.of())
+        return builder()
+                .setHiveMetastoreEndpoint(hiveMetastoreEndpoint)
+                .setS3Endpoint(s3Endpoint)
+                .setS3AccessKey(s3AccessKey)
+                .setS3SecretKey(s3SecretKey)
+                .setBucketName(bucketName)
+                .setHiveProperties(additionalHiveProperties)
                 .build();
     }
+
+    public static Builder builder(HiveMinioDataLake hiveMinioDataLake)
+    {
+        return builder()
+                .setHiveMetastoreEndpoint(hiveMinioDataLake.getHiveHadoop().getHiveMetastoreEndpoint())
+                .setS3Endpoint(hiveMinioDataLake.getMinio().getMinioApiEndpoint())
+                .setS3AccessKey(HiveMinioDataLake.ACCESS_KEY)
+                .setS3SecretKey(HiveMinioDataLake.SECRET_KEY)
+                .setBucketName(hiveMinioDataLake.getBucketName());
+    }
+
+    public static Builder builder()
+    {
+        return new Builder();
+    }
+
+    public static class Builder
+            extends HiveQueryRunner.Builder<Builder>
+    {
+        private HostAndPort hiveMetastoreEndpoint;
+        private HostAndPort s3Endpoint;
+        private String s3AccessKey;
+        private String s3SecretKey;
+        private String bucketName;
+
+        public Builder setHiveMetastoreEndpoint(HostAndPort hiveMetastoreEndpoint)
+        {
+            this.hiveMetastoreEndpoint = requireNonNull(hiveMetastoreEndpoint, "hiveMetastoreEndpoint is null");
+            return this;
+        }
+
+        public Builder setS3Endpoint(HostAndPort s3Endpoint)
+        {
+            this.s3Endpoint = requireNonNull(s3Endpoint, "s3Endpoint is null");
+            return this;
+        }
+
+        public Builder setS3AccessKey(String s3AccessKey)
+        {
+            this.s3AccessKey = requireNonNull(s3AccessKey, "s3AccessKey is null");
+            return this;
+        }
+
+        public Builder setS3SecretKey(String s3SecretKey)
+        {
+            this.s3SecretKey = requireNonNull(s3SecretKey, "s3SecretKey is null");
+            return this;
+        }
+
+        public Builder setBucketName(String bucketName)
+        {
+            this.bucketName = requireNonNull(bucketName, "bucketName is null");
+            return this;
+        }
+
+        @Override
+        public DistributedQueryRunner build()
+                throws Exception
+        {
+            requireNonNull(hiveMetastoreEndpoint, "hiveMetastoreEndpoint is null");
+            requireNonNull(s3Endpoint, "s3Endpoint is null");
+            requireNonNull(s3AccessKey, "s3AccessKey is null");
+            requireNonNull(s3SecretKey, "s3SecretKey is null");
+            requireNonNull(bucketName, "bucketName is null");
+
+            addHiveProperty("hive.s3.endpoint", "http://" + s3Endpoint);
```

Comment:
Can we require that `s3Endpoint` comes with `http://` prepended? Would be much better than just blindly adding it here

Reply:
s3Enpoint is passed as `HostAndPort` class. We are rendering it to string internally.

Reply:
> s3Enpoint is passed as `HostAndPort` class.

can we do the conversion to string and prepending with http:// as early as possible, so we don't need to deal with this here?

Reply:
We can. Then we should   probably do the same for hiveEndpoint.

Reply:
Changed s3Endpoint. Left hiveMetastoreEndpoint as `HostAndPort` matches usage of the latter better.